framework/src/main/java/org/checkerframework/common/wholeprograminference/SceneToStubWriter.java

```diff
@@ -0,0 +1,563 @@
+    }
+
+    /**
+     * The implementation of formatArrayType. Because of the (insane) structure of scenelib, the
+     * descent order is innermost array type to outermost array type, then component type. So if we
+     * have the type {@code @Foo int @Bar [] @Baz []}, the iteration order is {@code @Bar []}, then
+     * {@code @Baz []}, and then finally {@code @Foo int}. This implementation therefore passes a
+     * string builder with the result of the array types seen so far, to handle multidimensional
+     * arrays. That builder is appended to the final component type in the base case.
+     *
+     * @param e same as above
+     * @param arrayType same as above
+     * @param result the string builder containing the array types seen so far
+     * @return the formatted string, as above
+     */
+    private static String formatArrayTypeImpl(
+            ATypeElement e, String arrayType, StringBuilder result) {
+        String nextComponentType =
+                arrayType.indexOf('[') == -1
+                        ? null
+                        : arrayType.substring(0, arrayType.lastIndexOf('['));
+        // base case when the component is a non-array type
+        if (nextComponentType == null) {
+            return formatAnnotations(e.tlAnnotationsHere) + arrayType + " " + result.toString();
+        } else {
+            result.append(formatAnnotations(e.tlAnnotationsHere));
+            result.append("[] ");
+        }
+        // find the next array type; this loop should always find something, so throws an
+        // exception afterward if it fails
+        ATypeElement innerType = null;
+        for (Map.Entry<InnerTypeLocation, ATypeElement> ite : e.innerTypes.entrySet()) {
+            InnerTypeLocation loc = ite.getKey();
+            ATypeElement it = ite.getValue();
+            if (loc.location.contains(TypePathEntry.ARRAY)) {
+                innerType = it;
+            }
+        }
+
+        if (innerType == null) {
+            throw new BugInCF(
+                    "Encountered a multidimensional array without an inner type. This probably indicates a bug in the Annotation File Utilities.");
+        }
+
+        return formatArrayTypeImpl(innerType, nextComponentType, result);
+    }
+
+    /**
+     * Formats an AField so that it can be printed in a stub. An AField represents a variable
+     * declaration. In practice, {@code aField} should represent either a field declaration or a
+     * formal parameter of a method, because stub files should not contain local variable
+     * declarations. It is the responsibility of each caller of this method to ensure that is true.
+     *
+     * <p>It is also the responsibility of the caller to place the output of this method in context;
+     * because it is shared between field declarations and formal parameters, it does not add any
+     * trailing semicolons/commas/other syntax.
+     *
+     * @param aField the field to format
```

Comment:
This is confusing, because above the comment says that `aField` represents a field or formal.  Either say `AField` (but that's not helpful as it is obvious from the type) or "field or formal parameter".